lua/nvim-tree/git.lua

```diff
@@ -110,6 +147,81 @@ function M.update_status(entries, cwd)
     else
       node.git_status = nil
     end
+    ::continue::
+  end
+
+  if num_ignored > 0 and num_ignored == #entries then
+    parent_node.git_status = "ignored"
+  end
+end
+
+---A map from git roots to a list of ignored paths
+local gitignore_map = {}
+
+---Check if the given path is ignored by git.
+---@param path string Absolute path
+---@return boolean
+function M.should_gitignore(path)
+  for _, paths in pairs(gitignore_map) do
+    if paths[path] == true then
+      return true
+    end
+  end
+end
+
+---Updates the gitignore map if it's needed. Each entry in the map is only
+---updated if changes have been made to the git root's `.gitignore` or
+---`.git/info/exclude` files, or it's been invalidated by the
+---`invalidate_gitignore_map` function.
+function M.update_gitignore_map()
+  for git_root, git_status in pairs(roots) do
+    if git_status ~= not_git then
+      -- The mtime for `.gitignore` and `.git/info/exclude` is cached such that
+      -- the list of ignored files is only recreated when one of the said files
+      -- are modified.
+      for _, s in ipairs({".gitignore", ".git/info/exclude"}) do
+        local path = utils.path_join({git_root, s})
+        local stat = luv.fs_stat(path)
+        if stat and stat.mtime then
+          if not (fstat_cache[path]
+              and fstat_cache[path].mtime == stat.mtime.sec) then
+
+            gitignore_map[git_root] = {
+              _valid = false
+            }
+            fstat_cache[path] = {
+              mtime = stat.mtime.sec
+            }
+          end
+        end
+      end
+    end
+  end
+
+  for git_root, paths in pairs(gitignore_map) do
+    if not paths._valid then
+      gitignore_map[git_root] = {
+        _valid = true
+      }
+      paths = gitignore_map[git_root]
+
+      for _, s in ipairs(M.get_gitignored(git_root)) do
+        s = utils.path_remove_trailing(s, "/")
+        paths[utils.path_join({git_root, s}, "/")] = true
```

Comment:
is git also adding forward slashes on windows ? if not, you should use the `utils.path_separator` to do that.